packages/inferno-router/__tests__/loaderWithSwitch.spec.jsx

```diff
@@ -0,0 +1,298 @@
+    );
+
+    expect(container.innerHTML).toContain(TEXT);
+  });
+
+  it('Should only render one (1) component after click', async () => {
+    const [setDone, waitForRerender] = createEventGuard();
+    
+    const TEST = "ok";
+    const loaderFunc = async () => {
+      setDone();
+      return { message: TEST }
+    };
+
+    function RootComp() {
+      return <div id="root">ROOT</div>;
+    }
+
+    function CreateComp(props) {
+      const res = useLoaderData(props);
+      const err = useLoaderError(props);
+      return <div id="create">{res.message}</div>;
+    }
+
+    function PublishComp() {
+      return <div id="publish">PUBLISH</div>;
+    }
+
+    const tree = (
+      <MemoryRouter>
+        <div>
+          <nav>
+            <ul>
+              <li>
+                <NavLink exact to="/">
+                  Play
+                </NavLink>
+              </li>
+              <li id="createNav">
+                <NavLink to="/create">
+                  Create
+                </NavLink>
+              </li>
+              <li>
+                <NavLink to="/publish">
+                  Publish
+                </NavLink>
+              </li>
+            </ul>
+          </nav>
+          <Switch>
+            <Route exact path="/" component={RootComp} />
+            <Route path="/create" component={CreateComp} loader={loaderFunc} />
+            <Route path="/create" component={PublishComp} />
+          </Switch>
+        </div>
+      </MemoryRouter>
+    );
+
+    render(tree, container);
+
+    expect(container.innerHTML).toContain("ROOT");
+
+    // Click create
+    const link = container.querySelector('#createNav');
+    link.firstChild.click();
+
+    // Wait until async loader has completed
+    await waitForRerender();
+
+    expect(container.querySelector('#create').innerHTML).toContain(TEST);
+    expect(container.querySelector('#publish')).toBeNull();
+  });
+
+  it('can use a `location` prop instead of `router.location`', async () => {
+    const [setSwitch, waitForSwitch] = createEventGuard();
+    const [setDone, waitForRerender] = createEventGuard();
+    
+    const TEST = "ok";
+    const loaderFunc = async () => {
+      await waitForSwitch();
+      setDone();
+      return { message: TEST }
+    };
+
+    render(
+      <MemoryRouter initialEntries={['/one']}>
+        <NavLink id="link" to="/two">Link</NavLink>
+        <Switch>
+          <Route path="/one" render={() => <h1>one</h1>} />
+          <Route path="/two" render={(props) => {
+            const res = useLoaderData(props);
+            return <h1>{res.message}</h1>
+          }} loader={loaderFunc} />
+        </Switch>
+      </MemoryRouter>,
+      container
+    );
+
+    // Check that we are starting in the right place
+    expect(container.innerHTML).toContain('one');
+
+    const link = container.querySelector('#link');
+    link.click();
+
+    // TODO: I believe this test gives a false pass, in a demo, the page switches too early
```

Comment:
is this still an issue?

Reply:
The problem in the code is fixed, but the test passed before I implemented the fix. I'll see if I can make the test more accurate.

Reply:
Fixed in 625c5e0